salt/utils/gitfs.py

```diff
@@ -2262,27 +2270,39 @@ def clear_lock(self, remote=None, lock_type='update'):
             errors.extend(failed)
         return cleared, errors
 
-    def fetch_remotes(self):
+    def fetch_remotes(self, remotes=None):
         '''
         Fetch all remotes and return a boolean to let the calling function know
         whether or not any remotes were updated in the process of fetching
         '''
+        if remotes is None:
+            remotes = []
+        elif not isinstance(remotes, list):
+            log.error(
+                'Invalid \'remotes\' argument (%s) for fetch_remotes. '
+                'Must be a list of strings', remotes
+            )
+            remotes = []
+
         changed = False
         for repo in self.remotes:
-            try:
-                if repo.fetch():
-                    # We can't just use the return value from repo.fetch()
-                    # because the data could still have changed if old remotes
-                    # were cleared above. Additionally, we're running this in a
-                    # loop and later remotes without changes would override
-                    # this value and make it incorrect.
-                    changed = True
-            except Exception as exc:
-                log.error(
-                    'Exception caught while fetching %s remote \'%s\': %s',
-                    self.role, repo.id, exc,
-                    exc_info=True
-                )
+            name = getattr(repo, 'name', None)
+            if not remotes or (repo.id, name) in remotes:
```

Comment:
@terminalmage Does this actually works? Wouldn't this be instead:
```python
             if not remotes or repo.id in remotes or name in remotes:
```

Reply:
Or the `Must be a list of strings` line 2283 is incorrect.

Reply:
Good catch. No one uses this functionality to my knowledge, it was added for potential later use or for those using salt.utils.gitfs outside of the context of gitfs/git_pillar.

https://github.com/saltstack/salt/pull/53674